client/client_test.go

```diff
@@ -406,6 +406,6 @@ func TestPauses(t *testing.T) {
 
 	wg.Wait()
 	for _, e := range elapsed {
-		assert.True(t, e >= time.Second*time.Duration(retryDelaySeconds))
```

Comment:
Can you clarify why this fixes the test?

Reply:
@tigrannajaryan my best guess is that since we're comparing variable `e` which is of type `time.Duration` which is in nanoseconds, to time in seconds, there's something to do with accuracy. Maybe this comparison should be done in nanoseconds instead?

I tried running this test without my changes from the last PR and it is still flaky. Doing this comparison in Milliseconds seems to help address this flakiness.

Reply:
I can't tell from the source code what is the type/units of `e`. Obviously they need to be the same units for comparison to be meaningful. The code you have doesn't _look_ correct because you seem to be multiplying `retryDelaySeconds` which is in seconds to `time.Millisecond`. You need to make sure both values that are being compared are either `time.Time` or `uint64` nanoseconds (or whatever is the type for nanoseconds).
It is hard for me to confirm that this is a correct change just by looking at the code.